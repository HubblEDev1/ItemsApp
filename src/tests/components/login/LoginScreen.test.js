import { mount } from "enzyme";
import { MemoryRouter } from "react-router";
import { AuthContext } from "../../../auth/AuthContext";
import {LoginScreen} from '../../../components/login/LoginScreen';
import { types } from "../../../types/types";

describe('Testing <LoginScreen/>', () => {
    const history = {
        replace: jest.fn()
    }

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            name: 'Edwin',
            logged: true,
        }
    }

    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter>
                <LoginScreen history={history} />
            </MemoryRouter>
        </AuthContext.Provider>
    )


    test('should show it correctly', () => {
        console.log(wrapper.html());
        expect(wrapper).toMatchSnapshot();
    })


    test('should execute dispatch and navigation ', () => {
        wrapper.find('button').prop('onClick')()
        expect(history.replace).toHaveBeenCalled();
        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.login,
            payload :{
                name: 'Edwin'
            }
        });
    })
    
})
