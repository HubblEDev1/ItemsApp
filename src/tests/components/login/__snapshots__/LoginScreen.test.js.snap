// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <LoginScreen/> should show it correctly 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
